<a href="https://colab.research.google.com/github/cherishkim42/covid/blob/master/AmericaEDA.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
from google.colab import drive
drive.mount('/content/gdrive')

Go to this URL in a browser: https://accounts.google.com/o/oauth2/auth?client_id=947318989803-6bn6qk8qdgf4n4g3pfee6491hc0brc4i.apps.googleusercontent.com&redirect_uri=urn%3aietf%3awg%3aoauth%3a2.0%3aoob&response_type=code&scope=email%20https%3a%2f%2fwww.googleapis.com%2fauth%2fdocs.test%20https%3a%2f%2fwww.googleapis.com%2fauth%2fdrive%20https%3a%2f%2fwww.googleapis.com%2fauth%2fdrive.photos.readonly%20https%3a%2f%2fwww.googleapis.com%2fauth%2fpeopleapi.readonly

Enter your authorization code:
··········
Mounted at /content/gdrive


In [2]:
from scipy import stats
import math
import csv
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
sns.set(rc={'figure.figsize':(15,11)}) #standardize sizing for seaborn plots

/usr/local/lib/python3.6/dist-packages/statsmodels/tools/_testing.py:19: FutureWarning: pandas.util.testing is deprecated. Use the functions in the public API at pandas.testing instead.
  import pandas.util.testing as tm


# The US of A

### Questions to address:

1. What is the ratio of positive to negative tests?
1. What is the ratio of completed tests to pending tests?
1. Which states have the highest numbers of currently hospitalized?
1. What percentage of those admitted to hospitals are killed by the infection?

# What is the ratio of positive to negative tests?

In [6]:
usa = pd.read_csv("/content/gdrive/My Drive/Colab Notebooks/data/usa/us_covid19_daily.csv")
usa.head(10)

date  ...                                      hash
0  20200720  ...  67d367af6c685d8ba81dd2d4f9ea953c9e8a4d1e
1  20200719  ...  8ebbf82831bb2025b7603a8bf00ae9cbd7747175
2  20200718  ...  f27f83f9107b0cda5f154c50ba724c7df59b4ddd
3  20200717  ...  2fd5fa9462df840f6808d6757e4d13f702be8fdb
4  20200716  ...  8644aacf2b4997ff08daf87ae088c7f6a44cb367
5  20200715  ...  ef1be31b81541c1ff634f7adff6a5cebe14bab3e
6  20200714  ...  9fff665c0a4f59af0ca8746fa5d90e91b35c0967
7  20200713  ...  5b8ab51eb57a48502d534a29f1e5d62791ac8a24
8  20200712  ...  525f1c52fe9d66d1ba03e1417c6f2576aa9d9d4e
9  20200711  ...  a6708ca3148a9c8b5c9ca94686a9128f93f4a172

[10 rows x 25 columns]

The ratio of **3,813,916** positive to **42,655,608** negative reduces to **1 : 0.0894**

# What is the ratio of completed tests to pending tests?

In [9]:
print((3813916+ 42655608)/3003)

15474.366966366966


For every pending test, **15,474** have been completed. However, the data collected here did not take into account repeated tests; and someone who tests negative can later go out and get infected.

# Which states have the highest numbers of currently hospitalized?

In [16]:
states = pd.read_csv("/content/gdrive/My Drive/Colab Notebooks/data/usa/us_states_covid19_daily.csv")
states.head(3)

date state  positive  ...  positiveScore  score  grade
0  20200720    AK    2414.0  ...              0      0    NaN
1  20200720    AL   68891.0  ...              0      0    NaN
2  20200720    AR   33927.0  ...              0      0    NaN

[3 rows x 41 columns]

In [17]:
most_recent = states[(states['date'] == 20200720)]
most_recent.head(3)

date state  positive  ...  positiveScore  score  grade
0  20200720    AK    2414.0  ...              0      0    NaN
1  20200720    AL   68891.0  ...              0      0    NaN
2  20200720    AR   33927.0  ...              0      0    NaN

[3 rows x 41 columns]

In [22]:
(most_recent.sort_values('hospitalizedCurrently')).tail(10)

date state  positive  ...  positiveScore  score  grade
4   20200720    AZ  145183.0  ...              0      0    NaN
11  20200720    GA  145575.0  ...              0      0    NaN
5   20200720    CA  391538.0  ...              0      0    NaN
10  20200720    FL  360394.0  ...              0      0    NaN
47  20200720    TX  332434.0  ...              0      0    NaN
3   20200720    AS       0.0  ...              0      0    NaN
13  20200720    HI    1381.0  ...              0      0    NaN
18  20200720    KS   23334.0  ...              0      0    NaN
27  20200720    MP      37.0  ...              0      0    NaN
50  20200720    VI     304.0  ...              0      0    NaN

[10 rows x 41 columns]

The 5 states with the most currently hospitalized are:

1. TEXAS, **10,569**
1. FLORIDA, **9,489**
1. CALIFORNIA, **8,419**
1. GEORGIA, **3,183**
1. ARIZONA, **3,084**

# What percentage are killed by the infection?

In [23]:
print("California: ", (5183/21605))
print("Georgia: ", (3176/15047))
print("Arizona: ", (2784/6698))

California:  0.23989817171950936
Georgia:  0.2110719744799628
Arizona:  0.4156464616303374
